Fix event functions chain crash

sebjulliand opened the pull request:
### Changes
Resolves https://github.com/codefori/vscode-ibmi/issues/2181
This PR prevents a failing event function from crashing the whole chain (i.e. preventing other functions from being run when connecting/disconnecting).

### How to test this PR
1. Connect
2. Make sure every view loads correctly

Connecting to PUB400 should show a warning about the failing function.

### Checklist
* [x] have tested my change

worksofliam commented on src/api/Instance.ts:
```diff
@@ -66,7 +66,13 @@ export default class Instance {
+      }
+      catch(error){
+        const name = subscriber.func.name ? `Event function ${subscriber.func.name}` : `Anonymous event function`;
+        vscode.window.showWarningMessage(`${name} failed: ${error}`)
```
I don't know if the user cares. I think we should `console.error` here and let the using-extension handle the error in their function (e.g. show any warnings/issues to the user.)

sebjulliand replied:
Fair enough!

sebjulliand commented:
@worksofliam there you go.
I added a `subscribe` function that supersedes `onEvent` and replaced every occurrence of it.
I also added some time tracing around these function calls...it won't hurt, in case of a bottleneck situation.
And I also removed a few redundant view refreshes that were called twice.

worksofliam commented:
@sebjulliand Initial review of the code looks good. Glad to see `onEvent` marked as depreciated and replaced with `subscribe`. I will have a go at testing it tomorrow. I believe after this has merged, and with the release of 3.0.0, we will need to update the [documentation on our API](https://github.com/codefori/vscode-ibmi/pull/2182/files). After merging, and release of the next version, I can go ahead and try out `subscribe` in the Db2 for i extension.

worksofliam commented on src/api/Instance.ts:
```diff
@@ -56,17 +59,57 @@ export default class Instance {
+   */
   onEvent(event: IBMiEvent, func: Function): void {
-    this.subscribers.push({ event, func });
+    this.getSubscribers(event).set(`deprecated - ${func.name || "unknown"}_${this.deprecationCount++}`, { func });
```
I think an additional `console.log` in here so extension developers can see `onEvent` is going away in 3.0.0.

Something like `you are using onEvent, which is deprecated and will be removed in 3.0.0. Please use `subscribe`

sebjulliand replied:
Good idea!

worksofliam reviewed (approved):
I think one last change and we should be good to go here.

sebjulliand commented:
Tgabjs for the review @worksofliam . Give it a last whirl and I guess we'll be good to go!

worksofliam commented:
@sebjulliand I can confirm this PR fixed a startup bug with the database extension as the event error handler is working as expected.

<img width="796" alt="image" src="https://github.com/user-attachments/assets/20273c4e-ba81-414e-9622-3a1d4f9f5c3d">

worksofliam commented:
@sebjulliand Merge when you are ready! Thanks for your work on this.
